### ISSUE TITLE
0 connected peers for new node

### ISSUE THREAD

**Issue by garyyu** [mimblewimble/grin#1492]
With the latest code, a new build and run with clean `.grin` folder, can't start sync and 0 connected peers.

```



Sep 10 12:43:27.197 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:43:27.198 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:43:38.200 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:43:38.201 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:43:49.205 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:43:49.206 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }



Sep 10 12:45:17.398 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:45:17.400 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:45:28.404 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:45:28.404 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:45:39.405 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 109.74.202.16:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }
Sep 10 12:45:39.406 DEBG connect_peer: on 0.0.0.0:13414. Could not connect to 198.245.50.26:13414: Custom { kind: TimedOut, error: StringError("connection timed out") }

```
Tried a lot of times and always like that.

**Comment by garyyu** [comment]
After some investigation, several issues found:
1. default `seeding_type` is `DNSSeed`, up to now, we only have **one** dns seed available! for a stable Grin network, suggest at least 3 dns seeds available. Perhaps need ask for community help on this.
```
// DNS Seeds with contact email associated
const DNS_SEEDS: &'static [&'static str] = &[
   "t3.seed.grin-tech.org", // igno.peverell@protonmail.com
];
```

2. On the current one dns seed `t3.seed.grin-tech.org`, the dns query result is 3 socket addresses:
```
Sep 10 13:05:47.649 DEBG Retrieved seed nodes from dns: [V4(198.245.50.26:13414), V4(109.74.202.16:13414), V4(204.48.26.36:13414)]
```
The first 2 addresses have no connection (don't know who is running these 2 nodes and no way to address why), even with the 3 times retry. The seen problem is in the 3rd addresses, I saw the connection is success, but no response, and don't why there's no retry on this address, need more investigation.

**Comment by hashmap** [comment]
I see retries for all nodes:
```
Sep 10 09:43:55.190 DEBG connect_and_req: on 0.0.0.0:13414. connect to 204.48.26.36:13414 retrying 2
Sep 10 09:43:55.192 DEBG connect_and_req: on 0.0.0.0:13414. connect to 109.74.202.16:13414 retrying 2
Sep 10 09:43:55.193 DEBG connect_and_req: on 0.0.0.0:13414. connect to 198.245.50.26:13414 retrying 2
```
but all of them are down

**Comment by quentinlesceller** [comment]
My node is up @hashmap.

**Comment by garyyu** [comment]
3 dns seeds was replaced with one at 9th Jul.  by @ignopeverell

```
 const DNS_SEEDS: &'static [&'static str] = &[
-	"seed.grin-tech.org",      // igno.peverell@protonmail.com
-	"seed.grin.lesceller.com", // q.lesceller@gmail.com
-	"grin-seed.owncrypto.de",  // roll@yourowncryp.to
+	"t3.seed.grin-tech.org",   // igno.peverell@protonmail.com
 ];
```
could need add more back here.

**Comment by garyyu** [comment]
Just for info:
```
$ nslookup  t3.seed.grin-tech.org

Non-authoritative answer:
Name:	t3.seed.grin-tech.org
Address: 198.245.50.26
Name:	t3.seed.grin-tech.org
Address: 204.48.26.36
Name:	t3.seed.grin-tech.org
Address: 109.74.202.16

$ nslookup seed.grin.lesceller.com

Non-authoritative answer:
Name:    seed.grin.lesceller.com
Address: 198.245.50.26
Name:    seed.grin.lesceller.com
Address: 109.74.202.16
Name:    seed.grin.lesceller.com
Address: 46.4.91.48
Name:    seed.grin.lesceller.com
Address: 192.241.160.172

$ nslookup seed.grin-tech.org

Non-authoritative answer:
Name:	seed.grin-tech.org
Address: 46.4.91.48

$ nslookup grin-seed.owncrypto.de

Non-authoritative answer:
Name:	grin-seed.owncrypto.de
Address: 198.245.50.26
Name:	grin-seed.owncrypto.de
Address: 46.4.91.48
Name:	grin-seed.owncrypto.de
Address: 192.241.160.172
Name:	grin-seed.owncrypto.de
Address: 109.74.202.16
```

After removing the duplication, actually, the seed nodes are:
1. `198.245.50.26`
2. `109.74.202.16`
3. `204.48.26.36`
4. `192.241.160.172`

**Comment by garyyu** [comment]
And `seeding_type = "WebStatic"` don't work now, the file is not there:
```
$ wget http://grin-tech.org/seeds.txt
--2018-09-10 16:47:41--  http://grin-tech.org/seeds.txt
Resolving grin-tech.org (grin-tech.org)... 185.199.109.153, 185.199.110.153, 185.199.108.153, ...
Connecting to grin-tech.org (grin-tech.org)|185.199.109.153|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://grin-tech.org/seeds.txt [following]
--2018-09-10 16:47:41--  https://grin-tech.org/seeds.txt
Connecting to grin-tech.org (grin-tech.org)|185.199.109.153|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2018-09-10 16:47:41 ERROR 404: Not Found.
```

and Grin error:
```
Sep 10 16:46:39.372 DEBG Retrieving seed nodes from web http://grin-tech.org/seeds.txt
Sep 10 16:46:39.372 DEBG Started Dandelion transaction monitor.
Sep 10 16:46:39.372 WARN Grin server started.
Sep 10 16:46:41.044 ERRO
thread 'seed' panicked at 'Failed to resolve seeds: Error { inner:

Request error: Wrong response code }': libcore/result.rs:945stack backtrace:
   0:        0x10caf5bfe - backtrace::backtrace::trace::h16ebd892b2bc3b59
   1:        0x10caf4dbc - <backtrace::capture::Backtrace as core::default::Default>::default::hb901d1bdeee0be4c
   2:        0x10caf4e3d - backtrace::capture::Backtrace::new::h2b623e0f84041340
   3:        0x10ca86c91 - grin_util::logger::send_panic_to_log::{{closure}}::he64a95dc151f055c
   4:        0x10cc0efee - std::panicking::rust_panic_with_hook::h81c4f3add25e6667
   5:        0x10cc0ed6e - std::panicking::continue_panic_fmt::hfa057b7c1de88179
   6:        0x10cc0e9f8 - _rust_begin_unwind
   7:        0x10cc51a51 - core::panicking::panic_fmt::h4c82aa4e615d52a2
   8:        0x10c68492d - core::result::unwrap_failed::hef14d28be4a49610
   9:        0x10c721df0 - grin_servers::grin::seed::web_seeds::{{closure}}::h90f79f0400b8b6c0
  10:        0x10c720cd4 - grin_servers::grin::seed::connect_to_seeds_and_preferred_peers::h977caf3f7fdcd26a
  11:        0x10c67dbf3 - std::sys_common::backtrace::__rust_begin_short_backtrace::hf87ecd47a921039f
  12:        0x10c6d4387 - std::panicking::try::do_call::hef7e7c22ff381d2b
  13:        0x10cc1b07e - ___rust_maybe_catch_panic
  14:        0x10c6d89f4 - <F as alloc::boxed::FnBox<A>>::call_box::haa56ffdf67a98be1
  15:        0x10cc0df47 - std::sys_common::thread::start_thread::h78b1dd404be976ad
  16:        0x10cbf33e8 - std::sys::unix::thread::Thread::new::thread_start::h27c6becca8cf44e0
  17:     0x7fff51f0b660 - __pthread_body
  18:     0x7fff51f0b50c - __pthread_start


Sep 10 16:47:10.431 WARN sync: no peers available, disabling sync
Sep 10 16:47:10.431 DEBG sync_state: sync_status: Initial -> NoSync

```

**Comment by hendi** [comment]
Feel free to add `t3.grin-seed.prokapi.com` by `@hendi <info@prokapi.com>` which I'll always keep updated and pointing to several nodes. This is what I use to run grinexplorer.net

**Comment by ignopeverell** [comment]
This is still odd. My node wasn't up and had connected peers at that time.

**Comment by garyyu** [comment]
Some related peers issues can be tracked in https://github.com/mimblewimble/grin/issues/1464,  I would close this issue by #1496 .

But before closing this, please,  **who is the owner of `http://grin-tech.org/seeds.txt` file?**

Please update seeds.txt file and make it on line, if we still want to support the  **WebStatic** seed type.

**Comment by ignopeverell** [comment]
It's been removed a while ago. I think we should also remove `WebStatic` now.

**Comment by garyyu** [comment]
Okay, will submit a PR to remove `WebStatic`.

**Comment by garyyu** [comment]
closed by #1496 and #1512
